### ISSUE TITLE
Using the Officer package in a Multicore Parallel Environment

### ISSUE THREAD

**Issue by rishi-gupta-md** [davidgohel/officer#326]
Hi

I have installed the "parallel" package for multicore parallel processing of some functions which use the "officer" package.

The sample code is as follows:

```
library(officer)
library(parallel)

vector<- 1:100
docfun<-function(x){
  doc<-read_docx()
  doc<-body_remove(doc)
  title_format<-fp_text(bold = TRUE, font.size = 12,font.family = "Times New Roman",underlined = TRUE)
  title_para_format<-fp_par(padding.bottom = 12)
  title1<-fpar(ftext(as.character(x), prop=title_format),fp_p = title_para_format)
  doc<-body_add_fpar(doc,title1)
  return(doc)
 }
output_list<-parallel::mclapply(X=vector,FUN=docfun,mc.cores = 8)
```
When I run the last command, it gives a large list of 100 elements, but when I try and address any one of them, example "output_list[[1]]", it says "Error in xml_ns.xml_document(x) : external pointer is not valid"

the sample thing works well when I use the lapply function:
```
output_list<-lapply(X=vector,FUN=docfun)
```

Other functions work properly in the mclapply function. 
Am I missing something essential when applying the mclapply function to officer functions? or is there some other reason/solution for this?

Thanks in anticipation
Regards

**Comment by davidgohel** [comment]
Hi,

No sorry, it relies on an R6 object (an environnement) pointing to a temp file. So I think you can't use parallel processing and officer together if all cores should fill one single report.

**Comment by rishi-gupta-md** [comment]
Hello

Thank you for your response. I understand now why it is not working. Is it possible to incorporate this easily in the code?

Regards

**Comment by davidgohel** [comment]
If each node is producing one document, it is ok. They can then be inserted in a main document with function `body_add_docx`, see https://davidgohel.github.io/officer/reference/body_add_docx.html

**Comment by rishi-gupta-md** [comment]
Ok. Thank you.

**Comment by rishi-gupta-md** [comment]
Just an Update. I managed to get the multicore application working. Earlier I was making the mistake of returning the "doc" object, which I was then planning on combining.

What I did instead was write a word file within each worker function, and then just returned the filename, and the used the list of filenames which I then used to combine the word files. That worked perfectly!

Thanks for the direction.

Regards
